### PR TITLE
Update kubebuilder to commit b6a0bf1b6040

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
- Update kubebuilder dependency (See: https://github.com/kubernetes-sigs/kubebuilder/compare/9c02d557f01b...c158f4fa4207)
- Align SDK with Kubebuilder and address bug fixes

**Motivation for the change:**
- Solve tech-debt and keep both projects aligned

### CODE REVIEW

**Comment by jmrodri** [comment]
I retitled this PR it used to be called `Update kb 9c02d557f01b-to-b6a0bf1b6040` I changed it to `Update kubebuilder to commit b6a0bf1b6040`

**Review by jmrodri — APPROVED** [review]
/lgtm

**Comment by jberkhahn** [comment]
would be nice to wait until [1711](https://github.com/kubernetes-sigs/kubebuilder/pull/1711) is merged, but oh well :/

**Comment by camilamacedo86** [comment]
Hi @johananl, 

 > would be nice to wait until 1711 is merged, but oh well :/

I cannot see any no reason for we wait for any specific new fix or change in the Kubebuilder side.
See that the idea is we keep both aligned. If we do not keeping doing this update then, when we do that we will need to perform so many changes to bump that it will be very complex to get done.

**Comment by johananl** [comment]
@camilamacedo86 wrong tag ;-) Tagging @jberkhahn.

File: changelog/fragments/update-kb-from-9c02d557f01b-to-c158f4fa4207.yaml
Comment (estroz) [inline changelog/fragments/update-kb-from-9c02d557f01b-to-c158f4fa4207.yaml:6]: The "More info: [link]" string is too verbose. I say we stick to the `([kubebuilder#X](https://github.com/kubernetes-sigs/kubebuilder/pull/))` format, ex.
```suggestion
      (go/v2) Fixed controller imports scaffolded by `create api` when `--resource=false`
      ([kubebuilder#1770](https://github.com/kubernetes-sigs/kubebuilder/pull/1770))
```

File: internal/plugins/golang/v2/plugin.go
Comment (estroz) [inline internal/plugins/golang/v2/plugin.go:42]:
```suggestion
	_ plugin.Full          = Plugin{}
```

**Review by estroz — CHANGES_REQUESTED** [review]
Two nits, overall LGTM

File: internal/plugins/golang/v2/plugin.go
Comment (camilamacedo86) [reply internal/plugins/golang/v2/plugin.go:42]: Good cather 👍

**Review by estroz — APPROVED** [review]
/lgtm

**Pull request merged**
